Add MPAS-Analysis v0.4 (and more)

milenaveneziani opened the pull request:
This PR does a few things:
1) converts all csh scripts to bash. They are also all moved to a subdirectory called bash_scripts/
2) updates the MPAS-Analysis submodule with the v0.4 tag (which supports MPAS parallel tasking)
3) adds batch job scripts for edison (NERSC?)

milenaveneziani commented:
@kevans32, @xylar, @salilmahajan: as you can see, I did not exactly need 4 different scripts per machine, but only two. But let me start from the beginning.

We now have one single run script: run_aprime.bash. You can take a look at it and see if you like how I make it recognize the particular machine. Let me know if you have suggestions for improving this.

`run_aprime` sets a `run_batch_script` variable: if this is false, it simply calls `aprime_atm_diags` and `aprime_ocnice_diags` (our old ACME_atm_diags and ACME_ocnice_diags). If `run_batch_script` is true, then the batch scripts are submitted, which in turn call aprime_atm_diags and aprime_ocnice_diags. I have one batch script for atm and one for ocean.

I ran various tests on edison and things seem to be working. 
The one thing that puzzled me was that I needed to change setup_ocnice_config to remove the `run/` subdirectory for the namelist files `mpas-o_in` and `mpas-cice_in`. I am not sure why this is. The puzzling thing was that I still needed to specify the `run/` subdirectory for the stream files.. @xylar: maybe you understand the reason for this?

milenaveneziani commented:
Also, it would be great if Marcia (@mlbranstetter?) could take a look at the bash conversion scripts to make sure they are OK. The conversion took me longer that I had originally planned (of course..).

xylar commented on run_aprime.bash:
```diff
@@ -0,0 +1,325 @@
+machname4=`echo $HOSTNAME | cut -c1-4`
+machname5=`echo $HOSTNAME | cut -c1-5`
+machname6=`echo $HOSTNAME | cut -c1-6`
+if [ $machname6 == "edison" ]; then # NERSC
```
I would do this differently.  I would get rid of `machname#` and instead do:
```bash
if [ ${HOSTNAME:0:6} == "edison" ]; then
  machname="nersc"
elif [ ${HOSTNAME:0:4} == "rhea" ] || [ ${HOSTNAME:0:5} == "titan" ]; then
  machname="olcf"
elif [ ${HOSTNAME:0:5} == "aims4" ] || [ ${HOSTNAME:0:5} == "acme1" ]; then
  machname="llnl"
elif [ ${HOSTNAME:0:4} == "wolf" ]; then
  machname="lanl"
else
  echo "Unsupported host $HOSTNAME. Exiting."
  exit 1
if
```
It might make sense to use something like `centername` or `hpcname` instead of `machname` if you go this way. Then, I would use `machname` (or whatever you decide to call it) from here on (without 4, 5, 6).  From what I could see, there's no need to distinguish rhea from titan or aims4 from acme1 but keep those names separate if so.

milenaveneziani replied:
Thanks very much @xylar. I also didn't like to have the 4, 5, 6 numbers in there, but I looked for a way to find the short version of `hostname` and didn't find a unified way on all machines. I will try with your suggestions, and if those don't work, your code snip looks better than mine.

We may be able to remove the distinction between rhea and titan, but I'll keep aims4 and acme1 separate for now because their environments are not exactly identical.

xylar commented:
> I ran various tests on edison and things seem to be working.
The one thing that puzzled me was that I needed to change setup_ocnice_config to remove the run/ subdirectory for the namelist files mpas-o_in and mpas-cice_in. I am not sure why this is. The puzzling thing was that I still needed to specify the run/ subdirectory for the stream files.. @xylar: maybe you understand the reason for this?

Oops, I think this is a bug in MPAS-Analysis and shouldn't be changed here.  We should fix this and make a v0.41 tag.  (It's very bad practice to change an existing tag.  This is why I held off on making the v0.3 tag until we had nearly finished A-Prime testing the last time.)

xylar commented:
@milenaveneziani, I have created https://github.com/MPAS-Dev/MPAS-Analysis/pull/210 to fix the issue with the namelist path that you found.

xylar commented on python/setup_ocnice_config.py:
```diff
@@ -44,9 +44,9 @@ def check_env(envVarName):
+add_config_option(config, 'input', 'oceanNamelistFileName', 'mpas-o_in')
 add_config_option(config, 'input', 'oceanStreamsFileName', 'run/streams.ocean')
-add_config_option(config, 'input', 'seaIceNamelistFileName', 'run/mpas-cice_in')
+add_config_option(config, 'input', 'seaIceNamelistFileName', 'mpas-cice_in')
```
These changes should be reverted and instead the bug fix https://github.com/MPAS-Dev/MPAS-Analysis/pull/210 should be merged into MPAS-Analysis.

milenaveneziani replied:
ok, thanks. I will look at MPAS-Dev/MPAS-Analysis#210.

xylar commented on run_aprime.bash:
```diff
@@ -0,0 +1,325 @@
+      ./bash_scripts/aprime_ocnice_diags.bash
+    else
+      if [ $machname6 == "edison" ]; then # NERSC
+        sed -i "s/nodes=.*/nodes=$mpas_analysis_tasks/" ./bash_scripts/batch_ocnice.NERSC.bash
```
I don't think modifying `bash_scripts/batch_ocnice.NERSC.bash` in place is a good idea, since this is a file in the repository that might get committed with the changes.  Instead, `sed` should be creating a temporary copy of the bash script with the modification which is then run.  Probably the temporary script will need to be in the output directory (just like the config file) to prevent conflicts if analysis is run simultaneously on multiple simulations.

milenaveneziani replied:
you are right, I didn't think about that.
So, at the moment the config file still sits in the main a-prime directory, but it has a unique ID appended to it (the ID is created from the `date` command). I could do something similar for the batch script.

xylar commented on bash_scripts/batch_atm.NERSC.bash:
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -l
+#SBATCH --partition=regular
```
I don't think the atmospheric diagnostics are going to work as expected when they run on 1 processor on 1 node on the `regular` partition.  I think they will end up running in serial.

To take advantage of the way they are coded (i.e. launching several tasks and then waiting for them to complete), you will probably need to use the `shared` partition:
http://www.nersc.gov/users/computational-systems/edison/running-jobs/example-batch-scripts/#toc-anchor-13
I am not exactly sure how this will need to work, though.  My sense is that you will need to request more than one processor and maybe a certain amount of memory, as shown in one of the examples.  Anyway, I think this needs some work and testing for sure and should not be rushed!

milenaveneziani replied:
@salilmahajan: what is your opinion on this? Have you ever run the run_EDISON.sl batch script? That was what I took as an example to make `batch_atm.edison.bash`.

kevans32 commented:
I can see the ship has sailed on moving development to this public repo. @rljacob is it possible to make this repo private until the release of the code has been granted by DOE and we have the license headers etc in place.

milenaveneziani commented:
@kevans32: I understand your concern. But I would not like to keep developing on PreAndPostProcessingScripts because then I would have to redo the whole repo separation all over. Instead, I like your idea of making this repo private until we are ready for the public release.

milenaveneziani commented:
Now that we have v0.1 ready and released, it would be nice to get this in. Note that there are various conflicts that need to be solved, because of the license headers. I will take care of them, but they won't influence testing.

Also, on the MPAS-Analysis side, we are making progress on a number of things: adding ncclimo capability, startup documentation, ability to install it as conda package, and new T,S trend with depth diagnostics. This will likely be part of a v0.5 mpas-analysis tag that will come next for a-prime.

kevans32 commented:
@milenaveneziani if you want me to apply the license headers to all the new files introduced in this branch, it is not hard to do now that I have it figured out. Let me know when that would be a good time to do.

milenaveneziani commented:
Thanks @kevans32. Let me first check what these conflicts are in the first place, and make sure they are all related to missing headers.

milenaveneziani commented:
@kevans32: I have rebased and solved all the conflicts. Now this is ready for the license headers to be added. Would you mind giving it a try? Thanks.

kevans32 commented:
You want me to merge the updates from master into your branch?  Then run the script to add the headers to all the new files that do not yet have a header? Or is it not better for you to merge into master first, then we test that, add the license headers and merge that into master, to be consistent with what we have done before. But I can be convinced otherwise. Maybe it doesn't matter.

milenaveneziani commented:
I have already rebased to master, so the only thing left is to 1) test this, and for testing the license headers don't matter, of course, and 2) add the license headers. Since a-prime is officially public now, it seemed a good idea to add the license headers before merging this one.
But again, we can do 1) and 2) independently, and 1) is certainly most important.
What do you think?

milenaveneziani commented:
Basically, for 2) to happen before the merge, you would need to checkout this branch, add the license headers to all files again (I think only the generate_html bash script has it, for some reason), commit the changes and push them to this branch on github. Here are the steps, in case they are useful:
1) git fetch origin
2) git checkout -b add_mpasAnalysis_v0.4 origin/add_mpasAnalysis_v0.4
3) add changes
4) commit changes
5) check that all is good: git logg or git log
6) git push origin add_mpasAnalysis_v0.4

kevans32 commented:
@milenaveneziani sorry in fact the scripts and LICENSE files are there for existing files - I guess all the files are different so that is why at first I saw none with headers from the earlier version. I will proceed.

milenaveneziani commented:
exactly. All the csh files are gone, replaced by the bash files.
The python scripts are the same though, so those should still have the headers.

milenaveneziani commented:
> The python scripts are the same though, so those should still have the headers.

and indeed they do.

milenaveneziani commented:
thanks @kevans32.
I think we should also add the license header to this file also:
`python/setup_ocnice_config.py`
because this is part of a-prime not mpas-analysis (it's the file that gets the information from the run driver and puts it in a form that python can read and process).

kevans32 commented:
I am testing now, and getting the dreaded shared library error during remap. Standby- at least now I now how to sing any of the top 10 hits that cause this.

kevans32 commented:
I did not see that file has it missing. Hmm I will also look into that

kevans32 commented:
Ok- turns out I asked it not to include files called setup* because usually that is used for special files. So I took that out and committed.

Separate issue that flummoxes me: it seems that all the files are all committed with permissions mode 644 (rw,r,r) not 755 (Rex,rx,rx), which we need at least for most files to execute the scripts. For now I have just been changing it but it messes up a clean commit to have them in a different mode. Should I commit all the files to be 755 and not 644? Is everyone ok with that? If not please advise a better way.

milenaveneziani commented:
Could we change the permission mode for the bash only files? Those are the only ones we need executable mode.

kevans32 commented:
Sure, although I think we need to change more than just those since it calls several levels scripts that require execution privileges. Before I commit I want to make sure what I am doing is ok and will not cause problems. I did not know that git stores the mode of the files and that can change with a commit, so I wanted you guys to verify that I can change and commit this change to any files.

milenaveneziani commented:
I went back various commits to see where the permission change was introduced, and it looks like this was with the inclusion of the license headers. Did you mention that you use a script to add the headers to all files? Maybe that script changes the permissions as well?

milenaveneziani commented:
@kevans32: maybe we can alter the script that adds the license headers to change permissions to all bash files?

kevans32 commented:
Ok I am back to testing on rhea and loaded the right nco so no more library errors. I get a too many files error submitting in serial. Submitting in //, I get this in the job email it sends me:

moab_ERROR: partition 'login' does not exist
Error_Path: rhea8:/lustre/atlas/proj-shared/cli115/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs.logs/aprime_update_wwwdir.e2017-09-19_141823
Output_Path: rhea8:/lustre/atlas/proj-shared/cli115/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs.logs/aprime_update_wwwdir.o2017-09-19_141823

I feel like this is obvious- but why is it looking for a "login" partition? Should we not be on the compute nodes? Meantime I will test on titan to see if its a rhea thing.

kevans32 commented:
Running on titan: files get send to a new location I see, for ACME, instead of my home space so I do not have permission:
You don't have permission to access 
http://projects.olcf.ornl.gov/acme/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs/20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_RESTOM_ANN_reg_ts.png on this server.

I could access on master version of A-PRIME because it sent to my $HOME/www space. What did you guys do to get permission to see acme project space?

mlbranstetter commented:
I started testing run_aprime.bash on edison.  It seems to be having trouble finding a module called scipy.io.

milenaveneziani commented:
> Ok I am back to testing on rhea and loaded the right nco so no more library errors. I get a too many files error submitting in serial. 

hmm, we used to get the too many open files error on aims/acme only. not sure what changed.

>Submitting in //, I get this in the job email it sends me:
>
> moab_ERROR: partition 'login' does not exist
Error_Path: rhea8:/lustre/atlas/proj-shared/cli115/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs.logs/aprime_update_wwwdir.e2017-09-19_141823
Output_Path: rhea8:/lustre/atlas/proj-shared/cli115/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs.logs/aprime_update_wwwdir.o2017-09-19_141823

I have never seen this. Could you point me to your directory, as usual?

> I started testing run_aprime.bash on edison. It seems to be having trouble finding a module called scipy.io.

@mlbranstetter: could you check your python environment to be sure?:
```
which conda
conda info --envs
```
thanks.

milenaveneziani commented:
And, about the permissions for the html directory, I thought we had it in such a way that everybody on the cli115 project would have the right permissions automatically.. Let me double check this.

mlbranstetter commented:
which: no conda in (/usr/common/software/git/2.9.1/bin:.:/bin:/usr/common/software/darshan/3.1.4/bin:/usr/common/software/altd/2.0/bin:/usr/common/software/bin:/usr/common/mss/bin:/usr/common/nsg/bin:/opt/cray/pe/mpt/7.6.0/gni/bin:/opt/cray/rca/2.2.11-6.0.4.0_13.2__g84de67a.ari/bin:/opt/cray/alps/6.4.1-6.0.4.0_7.2__g86d0f3d.ari/sbin:/opt/cray/job/2.2.2-6.0.4.0_8.2__g3c644b5.ari/bin:/opt/cray/pe/craype/2.5.12.3/bin:/opt/intel/compilers_and_libraries_2017.2.174/linux/bin/intel64:/opt/cray/pe/modules/3.2.10.6/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin/X11:/usr/games:/usr/lib/mit/bin:/usr/lib/mit/sbin:/opt/cray/pe/bin:/global/u1/z/zender/bin)

So how do I get it to load conda?

milenaveneziani commented:
the right conda module should be loaded in `run_aprime`. You can search for this part:
```
if [ $machname == "nersc" ]; then
  module unload python
  module unload python_base
  module use $projdir/software/modulefiles/all
  module load python/anaconda-2.7-acme
...
```
so I am confused as to why that does not happen. Could you add some debugging statements in your run_aprime?
Also, if you point me to the location where you are running things from, I can take a look too.

kevans32 commented:
I noticed that we need to add documentation on all machines on how to load the environments. For rhea and titan I have scripts. Should we add these scripts to the a-prime repo for all to use?

milenaveneziani commented:
> I noticed that we need to add documentation on all machines on how to load the environments.

That should not be necessary, since we load the environment in `run_aprime`..

milenaveneziani commented:
@kevans32, about the permissions of the html directory, I see that your directory should have the right ones:
```
titan-ext5:proj_cli115/milena> ll /autofs/nccs-svm1_wwwproj/cli115/acme/
total 28K
drwxrwxr-x  6 root    cli115  4.0K Jul 19 16:38 ./
drwxrwxr-x  3 root    cli115  4.0K Jun 21 12:21 ../
drwxrwxr-x  7 4ue     4ue     4.0K Sep 19 14:07 4ue/
lrwxrwxrwx  1 root    root      37 Jun 21 16:48 acme -> /autofs/nccs-svm1_wwwproj/cli115/acme/
-rw-rw-r--  1 milena  cli115    12 Jun 21 12:21 index.html
drwxr-xr-x  6 milena  cli115  4.0K Jun 22 14:56 milena/
drwxrwxr-x 29 salil   salil   8.0K Sep  7 23:19 salil/
drwxrwxr-x 13 vanroek vanroek 4.0K Jul 25 10:01 vanroek/
```
but then, when I look into one of your run directories, I get this:
```
titan-ext5:proj_cli115/milena> ll /autofs/nccs-svm1_wwwproj/cli115/acme/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs/
total 27M
drwxrwxr-x 2 4ue 4ue     12K Sep 19 14:39 ./
drwxrwxr-x 7 4ue 4ue    4.0K Sep 19 14:07 ../
-rw------- 1 4ue 4ue    769K Sep 19 14:39 20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison-CERES-EBAF_FLUT_climo_ANN.png
...
```
which should give just you permission to view (so, not sure why you cannot see it also..).
I wonder if changing this line in run_aprime:
`chmod a+r $www_dir/$plots_dir_name`
to:
`chmod a+r $www_dir/$plots_dir_name/*`
would change things. 
Could you please give it a try?

xylar commented:
@milenaveneziani and @kevans32, it's not enough to give world read.  You also need world execute on the directory.  If there aren't any subdirectories, that should be easy enough.  If there are, you should do:
```bash
find $www_dir/$plots_dir_name -type d -exec chmod ugo+rx {} \;
find $www_dir/$plots_dir_name -type f -exec chmod ugo+r {} \;
```

milenaveneziani commented:
here is what I have in run_aprime now:
```
# Check on www_dir, permissions included
chmod a+rx $www_dir
# Create www_dir if it does not exist, purge it if it does
if [ ! -d $www_dir/$plots_dir_name ]; then
  mkdir $www_dir/$plots_dir_name
else
  rm -f $www_dir/$plots_dir_name/*
fi
chmod a+r $www_dir/$plots_dir_name
```
should it be changed to something similar to your command?

milenaveneziani commented:
Maybe like this?:
```
# Check on www_dir, permissions included
chmod a+rx $www_dir
# Create www_dir if it does not exist, purge it if it does
if [ ! -d $www_dir/$plots_dir_name ]; then
  mkdir $www_dir/$plots_dir_name
else
  rm -f $www_dir/$plots_dir_name/*
fi
find $www_dir/$plots_dir_name -type d -exec chmod ugo+rx {} \;
find $www_dir/$plots_dir_name -type f -exec chmod ugo+r {} \;
```

worleyph commented:
Unless there are items that you do not want to expose, you could use:

chmod -R a+rX $www_dir/$plots_dir_name

to change the directory and everything inside (recursively), or even just

chmod -R a+rX $www_dir

The capital X says to do the right then depending on whether this a file or a directory (as well as if there is an 's' specification).

milenaveneziani commented:
Thanks @worleyph.
@kevans32: let me know if one of those options solves the problem for you and I will adjust the script accordingly.

xylar commented:
@milenaveneziani, to be clear, the `chmod` should happend after the directory has been created and the contents are in place, not when you first create (or purge) the directory.

I agree @worleyph's suggestion is the best.  I didn't know about the X option.

milenaveneziani commented:
I will just change the first line to `chmod -R a+rX $www_dir` and remove the line after the if statement.

mlbranstetter commented:
Now I'm getting an error that no MPAS restart file was found:
MPAS-Analysis submodule:  e02b564202568c8e3ccdcfd4dc51fa3d8ea6f691 python/MPAS-Analysis (v0.43)
Traceback (most recent call last):
  File "python/MPAS-Analysis/run_analysis.py", line 197, in build_analysis_list
    analysisTask.setup_and_check()
  File "/global/u2/m/mbranst/a-prime/python/MPAS-Analysis/mpas_analysis/ocean/meridional_heat_transport.py", line 112, in setup_and_check
    self.simulationStartTime = get_simulation_start_time(self.runStreams)
  File "/global/u2/m/mbranst/a-prime/python/MPAS-Analysis/mpas_analysis/shared/timekeeping/utility.py", line 45, in get_simulation_start_time
    raise IOError('No MPAS restart file found: need at least one '
IOError: No MPAS restart file found: need at least one restart file for analysis to work correctly

There are files like mpaso.rst.am.timeSeriesStatsCustom.0004-01-01_00000.nc in the directory.

kevans32 commented:
@milenaveneziani  I seem to still have permissions issues and I altered the chmod in both places. 
Please test this yourself and see if you get the same issues. Currently I have these issues on each OLCF system:
1. Titan: permissions of plots (may get this on rhea if I get further)
2. on rhea, I get weird "moab_ERROR: partition 'login' does not exist" that probably needs to go to OLCF help. @worleyph have you ever seen this? Google has nothing to say

milenaveneziani commented:
@kevans32: I did a lot of testing on both rhea and titan, but that was a couple of months ago. I can try again now. I know for sure I don't have permissions problems with the html link, but I can investigate the issue when running parallel jobs.

@mlbranstetter: let me look into that.

milenaveneziani commented:
@kevans32: at OLCF, I can see the links you created if I open a page like this one:
http://projects.olcf.ornl.gov/acme/4ue/coupled_diagnostics_20161117.beta0.A_WCYCL1850S.ne30_oEC_ICG.edison_years16-20_vs_obs/
I get a permission access error if I just try to open this:
http://projects.olcf.ornl.gov/acme/4ue

I think you can change this behavior if you change the group of your directory:
`chgrp cli115 /autofs/nccs-svm1_wwwproj/cli115/acme4ue`
I also added this file in my directory, so that I can see a list of available cases when I open http://projects.olcf.ornl.gov/acme/milena:
`/autofs/nccs-svm1_wwwproj/cli115/acme/milena/index.php`
you are welcome to copy it over to your space if you'd like a similar behavior.

milenaveneziani commented:
@kevans32: I am investigating the issues with running the batch jobs on rhea. I see two problems:
1) the git submodule command times out:
```
ssh: connect to host github.com port 22: Connection timed out
fatal: The remote end hung up unexpectedly
Clone of 'git@github.com:MPAS-Dev/MPAS-Analysis.git' into submodule path 'python/MPAS-Analysis' failed
```
Therefore no ocn/ice plots are generated.
The atm plots are fine.

2) At OLCF I have a second batch script that runs on the login node and updates the `www_dir`: this was the best solution I could find and it was suggested by a very helpful OLCF consultant. For some reason, that is now throwing the `moab_ERROR: partition 'login' does not exist` message. It used to work before.. I will send another email to the consultants.

kevans32 commented:
I see same. Thanks

vanroekel commented:
@milenaveneziani git clone will not work on a batch job.  I'm fairly confident that compute nodes cannot access port 22.  When I tried this PR on OLCF I had to clone first on the compute node then submit the batch job, which ran fine.

milenaveneziani commented:
It did work in the summer though, when I first tested this..
Did you have this problem on titan as well?

milenaveneziani commented:
oh wait, maybe because I had already cloned it (instead now I am working with a new clone), and was just updating the sumodule??

vanroekel commented:
I think that is right.  When I had a cloned submodule already it worked
fine, but a fresh checkout of a-prime failed.  This was on titan.

On Tue, Sep 26, 2017 at 9:59 AM, Milena Veneziani <notifications@github.com>
wrote:

> oh wait, maybe because I had already cloned it (instead now I am working
> with a new clone), and was just updating the sumodule??
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ACME-Climate/a-prime/pull/3#issuecomment-332246172>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANB5Ueso3MjqnWVO1IpPRyvfp7KZI_hJks5smR88gaJpZM4N4DEx>
> .
>

milenaveneziani commented:
OK. I don't suppose there is a workaround to that. Does anyone know if there is? pinging @rljacob and @agsalin on this as well.
Basically, at least at OLCF, the fresh clone of a git submodule does not work within a batch job.

milenaveneziani commented:
On rhea, I also get a netcdf library error when running `ncremap`:
```
ncks: error while loading shared libraries: libnetcdf.so.11: cannot open shared object file: No such file or directory
ncremap: ERROR Failed to regrid. cmd_rgr[0] failed
ncpdq: error while loading shared libraries: libnetcdf.so.11: cannot open shared object file: No such file or directory
ncremap: ERROR Failed to generate pdq-file
```
I do **not** have this problem on titan.

(and I have @czender bin directory in my path (on both rhea and titan), but the run driver loads the conda module, which points to this nco version at the moment:
`nco                       4.6.5                         0    conda-forge`)

czender commented:
@milenaveneziani that problem will go away when you `conda update nco` to version 4.6.9 and run with `export NCO_PATH_OVERRIDE=No`, or if you don't use conda at all and point to my paths instead. Is there a reason not to conda update?

milenaveneziani commented:
yes, that is exactly what I was thinking. We are updating at OLCF now. Will check the other systems as well. And thanks for reminding me about setting `NCO_PATH_OVERRIDE`.

milenaveneziani commented:
All problems went away when we used the updated nco.

There are only two things left:
1) the git submodule issue when running in batch **and** the submodule is cloned for the first time. I will add instructions in the README file for this.

2) the issue with updating the `www_dir` in batch mode on rhea. I will disable running batch jobs on rhea for the time being. Batch mode works fine on titan.

I will add these changes shortly. After that and after @mlbranstetter's test on edison passes, we will be ready to merge.

@kevans32: I will let you know when that happens.

milenaveneziani commented:
ok. I have added the other small changes I mentioned above: tweaks in run_aprime and some documentation in README. I also changed the passwords for accessing the webpage on aims4/acme1.

This is ready to merge if @mlbranstetter's test on edison is successful (my various tests on edison went well in the past, but you never know how things can change).

milenaveneziani commented:
@mlbranstetter tests passed on edison as well.
Since @kevans32 is out of town and several people have done various testing of this, I will merge it.

sterlingbaldwin commented:
Now that this is finalized, what data does it require to run? I've using a pretty old version of coupled_diags, but the requirements before were:

    * raw cam.h0 history files (not regridded)
    * rpointer.atm and rpointer.ocn
    * mpas-cice_in and mpas-o_in
    * at least one mpaso.rst
    * streams.cice and streams.ocean
    * mpascice.hist.am.timeSeriesStatsMonthly.
    * mpaso.hist.am.timeSeriesStatsMonthly
 
Have any of these been removed/changed? Are you still doing your own regridding or is there an option to pass in already regridded cam.h0 files?

milenaveneziani commented:
@sterlingbaldwin: I have added a lot of information, including the files needed to run, in the header of the run_aprime driver. 
We don't need the rpointer files any longer.
MPAS does the regridding if the remapped/climatology files don't already exist.
I will leave the question for the cam regridding to @salilmahajan or @kevans32.

salilmahajan commented:
@sterlingbaldwin: A-Prime does its own regridding of cam.h0 files since we regrid each variable to the observational data grid that it is compared against, which are different for different variables.

rljacob commented:
@salilmahajan what tool do you use to do the regridding from cam.h0 files to observation grids?

salilmahajan commented:
@rljacob: we use ncremap.
